src/shoppinglist.cpp

```diff
@@ -0,0 +1,173 @@
+#include "shoppinglist.h"
+
+///Number of properties associated with each shopping list item (price, name, amount)
+const int NUM_SHOPPINGLIST_PROPERTIES = 3;
+///Column index for name in shopping list model
+#define ITEM_NAME_COL 0
+///Column index for price in shopping list model
+#define ITEM_PRICE_COL 1
+///Column index for amount in shopping list model
+#define ITEM_AMOUNT_COL 2
```

Comment:
I think the discussion on the exact display behavior can be moved to the PR for issue #8 when that becomes available, since no GUI behavior is, in principle, introduced in this PR. What's important for this PR is that the behavior of the (non-QAbstractTableModel-part of the) shopping list data handling makes sense, and that the access API is well-defined and well-named.